flink-libraries/flink-cep-scala/src/main/scala/org/apache/flink/cep/scala/package.scala

```diff
@@ -40,7 +44,11 @@ package object scala {
 
   private[flink] def cleanClosure[F <: AnyRef](f: F, checkSerializable: Boolean = true): F = {
     ClosureCleaner.clean(f, checkSerializable)
-    return f
+    f
+  }
+
+  private[flink] def mapToScala[T](map: JMap[String, JList[T]]): Map[String, Iterable[T]] = {
+    map.asScala.mapValues(_.asScala.toIterable).toMap
```

Comment:
Why not getting rid of the last `.toMap` and return a `collection.Map`? This will avoid an iteration over the elements of the `JMap` which is important given that this method will be invoked at every incoming element. From the rest, I think we are ok as the `asScala`s are just wrappers and the `.mapValues` are lazily evaluated.